From 83cc6bb11649c61afe2853b570de25c546462377 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Thu, 8 Mar 2018 14:43:34 +0200
Subject: [PATCH 3463/5242] MLK-17580: ASoC: fsl: sai: Use DSD helper

commit  222cb09ef07dcbcafd01ab3fcc8003de2b9e3efc from
https://source.codeaurora.org/external/imx/linux-imx.git

Replace DSD related code with calls to DSD helper functions.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_sai.c |   34 +++++++++++-----------------------
 sound/soc/fsl/fsl_sai.h |    2 +-
 2 files changed, 12 insertions(+), 24 deletions(-)

diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index ce15387..17194355 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -22,6 +22,7 @@
 #include <linux/mfd/syscon.h>
 #include <linux/mfd/syscon/imx6q-iomuxc-gpr.h>
 
+#include "fsl_dsd.h"
 #include "fsl_sai.h"
 #include "imx-pcm.h"
 
@@ -521,31 +522,21 @@ static int fsl_sai_hw_params(struct snd_pcm_substream *substream,
 	int ret;
 	int i;
 	int trce_mask = 0;
-	snd_pcm_format_t format = params_format(params);
 
 	if (sai->slots)
 		slots = sai->slots;
 
 	pins = DIV_ROUND_UP(channels, slots);
+	sai->is_dsd = fsl_is_dsd(params);
+	sai->pins_state = fsl_get_pins_state(sai->pinctrl, params);
 
-	if (format == SNDRV_PCM_FORMAT_DSD_U8 ||
-		format == SNDRV_PCM_FORMAT_DSD_U16_LE ||
-		format == SNDRV_PCM_FORMAT_DSD_U16_BE ||
-		format == SNDRV_PCM_FORMAT_DSD_U32_LE ||
-		format == SNDRV_PCM_FORMAT_DSD_U32_BE) {
-		sai->is_dsd = true;
-
-		if (!IS_ERR_OR_NULL(sai->pins_dsd)) {
-			ret = pinctrl_select_state(sai->pinctrl, sai->pins_dsd);
-			if (ret) {
-				dev_err(cpu_dai->dev,
-					"failed to set proper pins state: %d\n", ret);
-				return ret;
-			}
+	if (!IS_ERR_OR_NULL(sai->pins_state)) {
+		ret = pinctrl_select_state(sai->pinctrl, sai->pins_state);
+		if (ret) {
+			dev_err(cpu_dai->dev,
+				"failed to set proper pins state: %d\n", ret);
+			return ret;
 		}
-	} else {
-		pinctrl_pm_select_default_state(cpu_dai->dev);
-		sai->is_dsd = false;
 	}
 
 	if (sai->is_dsd)
@@ -918,8 +909,8 @@ static int fsl_sai_dai_resume(struct snd_soc_dai *cpu_dai)
 	struct fsl_sai *sai = snd_soc_dai_get_drvdata(cpu_dai);
 	int ret;
 
-	if (sai->is_dsd && !IS_ERR_OR_NULL(sai->pins_dsd)) {
-		ret = pinctrl_select_state(sai->pinctrl, sai->pins_dsd);
+	if (!IS_ERR_OR_NULL(sai->pins_state)) {
+		ret = pinctrl_select_state(sai->pinctrl, sai->pins_state);
 		if (ret) {
 			dev_err(cpu_dai->dev,
 				"failed to set proper pins state: %d\n", ret);
@@ -1346,9 +1337,6 @@ static int fsl_sai_probe(struct platform_device *pdev)
 
 	sai->pinctrl  = devm_pinctrl_get(&pdev->dev);
 
-	if (!IS_ERR_OR_NULL(sai->pinctrl))
-		sai->pins_dsd = pinctrl_lookup_state(sai->pinctrl, "dsd");
-
 	platform_set_drvdata(pdev, sai);
 
 	pm_runtime_enable(&pdev->dev);
diff --git a/sound/soc/fsl/fsl_sai.h b/sound/soc/fsl/fsl_sai.h
index 7728c70..84f75fe 100644
--- a/sound/soc/fsl/fsl_sai.h
+++ b/sound/soc/fsl/fsl_sai.h
@@ -213,7 +213,7 @@ struct fsl_sai {
 	const struct fsl_sai_soc_data *soc;
 	struct pm_qos_request pm_qos_req;
 	struct pinctrl *pinctrl;
-	struct pinctrl_state *pins_dsd;
+	struct pinctrl_state *pins_state;
 };
 
 #define TX 1
-- 
1.7.9.5

